test/e2e/apimachinery/aggregator.go

```diff
@@ -570,3 +577,42 @@ func generateFlunderName(base string) string {
 	}
 	return fmt.Sprintf("%s-%d", base, id)
 }
+
+func waitAPICoherentResponse(f *framework.Framework) {
+	framework.Logf("Waiting on v1alpha1.wardle.example.com to disappear from all apiserver instances")
+	client := f.ClientSet
+
+	endpoints, err := client.CoreV1().Endpoints("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+	framework.ExpectNoError(err, "getting kubernetes endpoints in namespace default")
+	var apiPort int32 = 0
+	apiAddresses := make([]string, 0)
+	for _, subset := range endpoints.Subsets {
+		for _, port := range subset.Ports {
+			if port.Name != "https" || port.Protocol != "TCP" || port.Port == 0 {
+				continue
+			}
+			apiPort = port.Port
+		}
+		for _, address := range subset.Addresses {
+			apiAddresses = append(apiAddresses, address.IP)
+		}
+	}
+	if len(apiAddresses) == 0 || apiPort == 0 {
+		framework.Failf("failed to get apiserver ip addresses or port")
+	}
+	execPod := e2epod.CreateExecPodOrFail(f.ClientSet, f.Namespace.Name, "verify-apiserver-response", nil)
+	for _, address := range apiAddresses {
+		cmd := fmt.Sprintf(`curl -s -o /dev/null -w '%%{http_code}' https://kubernetes:%d/apis/wardle.example.com/v1alpha1 --resolve kubernetes:%d:%s --cacert %s -H "Authorization: Bearer $(cat %s)"`,
+			apiPort,
+			apiPort,
+			address,
+			"/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+			"/var/run/secrets/kubernetes.io/serviceaccount/token",
+		)
+		stdout, err := framework.RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		framework.ExpectNoError(err, "failed to execute curl command")
+		if stdout != "404" {
+			framework.Failf("Received status code %s instead of 404", stdout)
+		}
```

Comment:
this will not solve the flakiness, if you are "waiting" for a 404, you should run this in a loop and fail if the loop never gets the 404
```
err :=wait.PollImmadiate(period, timeout, func() bool, error {

output , err := execPod(cmd)
if stdout != 404 {
   // retry
   return false, nil
}
// 404 received, move one
return true, nil
})
if err != nil {
framework.Failf("failed to get a 404 on address %s)
}
```

Reply:
Must be feeling funny today, I miswrote the whole point of the test. You are right it should be enclosed within a `wait.PollImmediate`.